naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/middleware/MetricsCollectionMiddleware.scala

```diff
@@ -39,14 +39,27 @@ class MetricsCollectionMiddleware(
 
 trait GraphQLMetricsCollector {
   def markFieldError(fieldName: String): Unit
+  def timeQueryParsing[A](f: => A): A
 }
 
 class LoggingMetricsCollector extends GraphQLMetricsCollector with StrictLogging {
   def markFieldError(fieldName: String): Unit = {
     logger.info(s"Error when loading field $fieldName")
   }
+
+  def timeQueryParsing[A](f: => A): A = {
+    val before = System.currentTimeMillis()
+    val res = f
+    val after = System.currentTimeMillis()
+    logger.info(s"Parsed query in ${after - before}ms")
```

Comment:
This message would be much more useful if you passed in and logged `operation` here.  Maybe some graphQL queries are slower than others.

Reply:
Another note about this.  This is going to probably cause a lot of logging.  Consider logging at `debug` level, then we turn it on while triaging?  I'm not sure how you guys normally do this.

Reply:
This MetricsCollector isn't designed to be used in production (we override this with a DatadogMetricsCollector in our repo), so this is mostly just used for debugging in the `example` project

Reply:
ah, but I could tag the metric with operation name. good point